File: exercise/python/day24/domain/stock_reduced_event.py
import datetime
import uuid

from domain.core.event import Event
from domain.stock_unit import StockUnit

class StockReducedEvent(Event):
    def __init__(self, event_id: uuid.UUID, date: datetime, toy_name: str, new_stock: StockUnit):
        super().__init__(event_id, date)
        self.toy_name = toy_name
        self.new_stock = new_stock

File: exercise/ts/day24/src/domain/StockReducedEvent.ts
import {Event} from "./core/Event"
import {StockUnit} from "./StockUnit";

export class StockReducedEvent extends Event {
    constructor(
        id: string,
        date: Date,
        public toyName: string,
        public newStock: StockUnit
    ) {
        super(id, 1, date);
    }
}
